packages/services/f-wdio-utils/src/page.object.js

```diff
@@ -0,0 +1,24 @@
+class Page {
+    constructor () {
+        this.title = 'Component URLS';
+    }
+
+    openOrganism (path) {
```

Comment:
This could probably be a single function that utilises a switch statement
```
openComponent(componentType, path) {
    let formattedUrl = ''
    
    switch(componentType.toLowerCase()) {
      case atom:
        formattedUrl = `/iframe.html?id=components-atoms--${path}`
        break;
      case molecule:
        formattedUrl = `/iframe.html?id=components-molecules--${path}`
        break;
      case organism:
        formattedUrl = `/iframe.html?id=components-organisms--${path}`
        break;
      default:
        throw new Error(`${componentType} is not a valid component type. Please use 'atom', 'molecule', or 'organism'`);
    }
    
    browser.url(formattedUrl)
}
```

Reply:
ooh yeah could do! I guess this way means having to specify the type too though? Would it be quicker? Is it better to have them all in one? I'm not sure which way I prefer 😄

Reply:
Whilst this solution is more complex, from the perspective of a user writing a new test in a spec file, you don't want to have to think about what type a component is.

We'll be able to extend this in the future to something like:


```

// f-checkout.component.js

let url = '?path=/story/components-organisms--checkout-component'

withLocale(locale) {
    url = `${url}&knob-locale=${locale}`
    
    return this;
}

withCollection() {
    url = `${url}&knob-Checkout%20Url=%2Fcheckout-collection.json`
    
    return this;
}

openComponent() {
    super.openComponent('organism', url);   
}
```


Then in the spec file we can do something like:

```
f-checkout-collection.spec.js


....

    beforeEach(() => {
         CheckoutComponent.withLocale('en-AU').withCollection();
         CheckoutComponent.openComponent()
         CheckoutComponent.open.waitForComponent();
     });
```


With this approach, the spec file doesn't have to worry about what function to call :)

Hope that makes sense, but this is my thinking atm as to how we could simply this going forward!

Reply:
yeah definitely agree! It makes sense and will be so much nicer and more organised! 😄